crates/protocol/src/packets/client/handshake.rs

```diff
@@ -0,0 +1,54 @@
+use std::io::Cursor;
+
+use super::*;
+use crate::{io::Decoder, Encoder};
+
+packets!(
+    Handshake {
+        protocol_version VarInt;
+        server_address String;
+        server_port u16;
+        next_state HandshakeState;
+    }
+);
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
```

Comment:
Deriving `PartialOrd` and `Ord` may be reasonable here since the states have an explicit relation.

Reply:
Who's going to put these enums in a sorted array?

Reply:
It may be useful, e.g. when comparing the order of player states (e.g., validating that the currently requested state `>` previos state).